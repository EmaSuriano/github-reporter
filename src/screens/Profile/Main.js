import React, { Component } from "react";
import { isEmpty } from "ramda";
import { Box, Grid } from "grommet";

import Header from "./components/Header";
import User from "./components/User";

import "./styles.css";

export default class Profile extends Component {
  state = { user: "" };

  searchProfile = user => this.setState({ user });

  render() {
    const {
      searchProfile,
      state: { user }
    } = this;

    return (
      <Grid
        areas={[
          { name: "header", start: [0, 0], end: [1, 1] },
          { name: "sidebar", start: [0, 1], end: [0, 1] },
          { name: "main", start: [1, 1], end: [1, 1] }
        ]}
        fill
        rows={["flex", "flex"]}
        columns={["flex", "flex"]}
      >
        <Box
          basis="medium"
          columns={["large", "auto"]}
          gridArea="header"
          direction="row"
          align="center"
          justify="start"
          pad={{ horizontal: "medium", vertical: "small" }}
        >
          <Header searchProfile={searchProfile} />
        </Box>
        <Box
          gridArea="sidebar"
          width="small"
          animation={[
            { type: "fadeIn", duration: 300 },
            { type: "slideRight", size: "xlarge", duration: 150 }
          ]}
        >
          Second Box
        </Box>
        <Box gridArea="main" justify="center" align="center">
          Third Box
        </Box>
      </Grid>
    );

    /* return (
      <section className="container">
        <Header searchProfile={searchProfile} />
        {!isEmpty(user) && <User profile={user} />}
      </section>
    ); */
  }
}
