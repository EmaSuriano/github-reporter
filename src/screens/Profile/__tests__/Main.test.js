import React from "react";
import { shallow } from "enzyme";

import Main from "../Main";

describe("Main Profile Screen Component", () => {
  let wrapper;

  describe("render", () => {
    describe("when user is empty", () => {
      beforeEach(() => {
        wrapper = shallow(<Main />);
      });

      it("should match with the html structre of the snapshot", () =>
        expect(wrapper).toMatchSnapshot());
    });

    describe("when user has data", () => {
      beforeEach(() => {
        wrapper = shallow(<Main />);
        wrapper.setState({ user: "wesbos" });
      });

      it("should match with the html structre of the snapshot", () =>
        expect(wrapper).toMatchSnapshot());
    });
  });

  describe("inner methods", () => {
    beforeEach(() => {
      wrapper = shallow(<Main />);
    });

    describe("searchProfile", () => {
      let user;

      beforeAll(() => {
        wrapper.instance().searchProfile("kentcdodds");
        user = wrapper.state("user");
      });

      it("should modify the state when call to searchProfile", () =>
        expect(user).toBe("kentcdodds"));
    });
  });
});
