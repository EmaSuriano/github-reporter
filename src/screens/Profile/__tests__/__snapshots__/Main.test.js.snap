// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Profile Screen Component render when user has data should match with the html structre of the snapshot 1`] = `
<section
  className="container"
>
  <Header
    searchProfile={[Function]}
  />
  <User
    profile="wesbos"
  />
</section>
`;

exports[`Main Profile Screen Component render when user is empty should match with the html structre of the snapshot 1`] = `
<section
  className="container"
>
  <Header
    searchProfile={[Function]}
  />
</section>
`;
